### ISSUE TITLE
Possible memory leak with BackgroundExecutor using BackgroundExecutorWrapper ThreadLocals (for scheduled tasks)

### ISSUE THREAD

**Issue by rPraml** [ebean-orm/ebean#2854]
Hello Rob, we introduced with https://github.com/ebean-orm/ebean/pull/2590 a concept, that each operation that is executed by the BackgroundExecutor may inherit ThreadLocals by the originating thread

## Expected behavior

when a runnable/callable will be submitted to the executor, ThreadLocals are available in the async thread.
After execution, the memory should be free-ed.

## Actual behavior

The ThreadLocals are also copied to periodic threads and will stay there forever.

### Investigation

I discovered this issue, because I was wondering why a certain user session was not garbage collected.

What happened:

We store the user session in a ThreadLocal, so that it is always available in the current thread.
This ThreadLocal acts also as CurrentTenant/CurrentUserProver. We also have a BackgroundExecutorWrapper that copies this information.

When a User now accesses a cache the first time, a ServerCache is created with computeIfAbsent [here](https://github.com/ebean-orm/ebean/blob/master/ebean-core/src/main/java/io/ebeaninternal/server/cache/DefaultCacheHolder.java#L77)

The cache registers a periodicTrim-task in the BackgroundExecutor and the BackgroundExecutorWrapper will copy the current ThreadLocal (where it stays forever)

long story short: @rbygrave What would you suggest, how to fix it
- only wrap "one-time-tasks" that are executed or submitted (IMHO it is not needed - in our use cases - to copy ThreadLocals to periodic tasks)
- add a boolean flag to BackgroundExecutorWrapper.wrap that indicates, if this is a periodic task. The code in the wrapper can then decide, if it is necessary to copy threadLocals
- add a BackgroundExecutorWrapper.wrapPeriodically

I can create a PR when I get feedback

**Comment by rbygrave** [comment]
> only wrap "one-time-tasks"

I agree. Yes, I think this is the change / fix we should go with.

> not needed - in our use cases - to copy ThreadLocals to periodic tasks

Yes and I believe that is **_generally_** true.  Scheduled tasks by their nature should not "inherit" ThreadLocals - I think that makes good sense to me. Lets make it so.

> I can create a PR when I get feedback

Awesome, thanks!!
